### PR TITLE
readme: Add build instructions for Haiku

### PR DESCRIPTION
**Pull request by stenstorp**
Since there's a package for Haiku, may as well add instructions for it.

### CODE REVIEW

**Comment by stenstorp** [comment]
Sorry about the close/open, thought the "Merging is blocked" thing meant I did something wrong, haha.

**Comment by AJenbo** [comment]
Much appriciated, I hadn't been able to install the dependencies (got dependency errors with broken chains using Depo, both on beta and alpha), so coudn't confirm the instructions or make a release build.

**Comment by stenstorp** [comment]
Yeah, the repos can be a bit slow/connection issues. You're also probably better off on the nightlies anyway, that's what most people use.

**Comment by AJenbo** [comment]
So this is the problem I keep hitting:
image

File: README.md [hunk 1]
```diff
@@ -145,6 +145,20 @@ cmake -DCMAKE_C_COMPILER=/usr/local/bin/gcc8 -DCMAKE_CXX_COMPILER=/usr/local/bin
 make -j$(sysctl -n hw.ncpu)
 ```
 </details>
+<details><summary>Haiku</summary>
+
+### Installing dependencies
+```
+pkgman install cmake devel:libsdl2 devel:libSDL2_mixer devel:libSDL2_ttf devel:libsodium
```
Comment (AJenbo) [inline README.md]: A fresh nightly appears to want the names in lowercase
```suggestion
pkgman install cmake devel:libsdl2 devel:libsdl2_mixer devel:libsdl2_ttf devel:libsodium
```

Comment (stenstorp) [reply README.md]: Of course it is. Whoops.

Comment (stenstorp) [reply README.md]: Also, looks like 32 bit Haiku puts "_x86" after package names. Something to do with using different versions of gcc for BeOS compatibility...

**Comment by stenstorp** [comment]
32 bit Haiku is being a bit of an issue, going to have to look into that a bit more.

**Comment by AJenbo** [comment]
Interesting. At the moment I can only get 32bit up and running as the 64bit image fails with a message about not being able to detect a file system :(

**Comment by stenstorp** [comment]
Looks like 32 bit Haiku can't build the debug, complains about libsodium not having debug stuff.
That package install error you're getting seems to be an issue with the official repo, didn't happen when I used a mirror (cdn.haiku.nz).
32 bit Haiku also seems to want to use gcc 2.95 by default, gcc-x86/g++-x86 is version 7.3.0.

**Pull request merged**